webapp/components/NotificationsTable/NotificationsTable.vue

```diff
@@ -61,41 +83,36 @@ export default {
     notifications: { type: Array, default: () => [] },
   },
   computed: {
+    ...mapGetters({
+      currentUser: 'auth/user',
+    }),
     fields() {
       return {
         icon: {
           label: ' ',
-          width: '5',
-        },
-        user: {
-          label: this.$t('notifications.user'),
-          width: '45%',
+          width: '3%',
         },
-        post: {
-          label: this.$t('notifications.post'),
+        triggerer: {
+          label: ' ',
           width: '25%',
         },
-        content: {
-          label: this.$t('notifications.content'),
+        title: {
+          label: ' ',
           width: '25%',
         },
+        metadata: {
+          label: ' ',
+          width: '47%',
+        },
       }
     },
+    notificationsData() {
+      return this.notifications.map(notification =>
+        extractNotificationDataOfCurrentUser(notification, this.currentUser),
+      )
+    },
```

Comment:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2020-01-23T19:31:31Z" title="Thursday, January 23rd 2020, 8:31:31 pm +01:00">Jan 23, 2020</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -61,41 +96,38 @@ export default {
     notifications: { type: Array, default: () => [] },
   },
   computed: {
+    ...mapGetters({
+      currentUser: 'auth/user',
+    }),
     fields() {
       return {
         icon: {
           label: ' ',
-          width: '5',
+          width: '3%',
         },
         user: {
-          label: this.$t('notifications.user'),
-          width: '45%',
+          label: ' ',
+          width: '25%',
         },
         post: {
-          label: this.$t('notifications.post'),
+          label: ' ',
           width: '25%',
         },
         content: {
-          label: this.$t('notifications.content'),
-          width: '25%',
+          label: ' ',
+          width: '47%',
         },
       }
     },
+    notificationsData() {
+      const data = []
+      this.notifications.forEach(notification => {
+        data.push(extractNotificationDataOfCurrentUser(notification, this.currentUser))
+      })
```


```suggestion
      return this.notifications.map(notification => {
        extractNotificationDataOfCurrentUser(notification, this.currentUser)
      })
 ```
```

Reply:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2020-01-23T19:31:45Z" title="Thursday, January 23rd 2020, 8:31:45 pm +01:00">Jan 23, 2020</time>_

---


I think this should work, oder?

Reply:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2020-02-01T14:45:01Z" title="Saturday, February 1st 2020, 3:45:01 pm +01:00">Feb 1, 2020</time>_

---


Hell, yes. That's one of the exam questions for my students: How to eliminate unnecessary temporary states like `data` in this case..

Reply:
> [<img alt="Tirokk" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/Tirokk) **Authored by [Tirokk](https://github.com/Tirokk)**
_<time datetime="2020-02-05T11:37:06Z" title="Wednesday, February 5th 2020, 12:37:06 pm +01:00">Feb 5, 2020</time>_

---


You are really close to it. @mattwr18 
Just leave the curly braces away or put a `return` in front of `extractNotificationDataOfCurrentUser `.

@roschaefer Now I do really understand `eliminate unnecessary temporary states`!